CHANGELOG.md

```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## Unreleased
+## [Unreleased]
```

Comment:
`[]` needs a hyperlink somewhere

Reply:
[]() is used to describe a link but does it mean every [] must be a link? What if you just want to write `[someword]`? :)

Following [keepachangelog](https://keepachangelog.com/en/1.0.0/) here so the changelog library works with our changelog. Without `[]`, it is not able to parse correctly.

Reply:
Yeah, but then the hyperlinks are at the end of the file 😉

Reply:
Here is an example https://raw.githubusercontent.com/signalfx/splunk-otel-go/main/CHANGELOG.md

Reply:
I'm a bit confused. Are you saying it we enclose version numbers in [] like I did in this PR then we MUST add links at the bottom? What happens if we don't? Everything seems to work fine to me 😃

Or are you suggesting we should add it as an improvement? In that case, we can have a different PR to improve changelog format.

Reply:
Suggesting to add as an improvement